pkgs/initiator/initiator.go

```diff
@@ -0,0 +1,790 @@
+	resc := make(chan opReqResult, len(operatorsIDs))
+	for _, op := range operatorsIDs {
+		go func(operator Operator) {
+			res, err := c.SendAndCollect(operator, method, msg)
+			resc <- opReqResult{
+				operatorID: operator.ID,
+				err:        err,
+				result:     res,
+			}
+		}(c.Operators[op.ID])
+	}
+	final := make([][]byte, 0, len(operatorsIDs))
+
+	errarr := make([]error, 0)
+
+	for i := 0; i < len(operatorsIDs); i++ {
+		res := <-resc
+		if res.err != nil {
+			errarr = append(errarr, res.err)
+			continue
+		}
+		final = append(final, res.result)
+	}
+
+	finalerr := error(nil)
+
+	if len(errarr) > 0 {
+		finalerr = errors.Join(errarr...)
+	}
+
+	return final, finalerr
+}
+
+func parseAsError(msg []byte) (error, error) {
+	sszerr := &wire.ErrSSZ{}
+	err := sszerr.UnmarshalSSZ(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return errors.New(string(sszerr.Error)), nil
+}
+
+func (c *Initiator) VerifyAll(id [24]byte, allmsgs [][]byte) error {
+	for i := 0; i < len(allmsgs); i++ {
+		msg := allmsgs[i]
+		tsp := &wire.SignedTransport{}
+		if err := tsp.UnmarshalSSZ(msg); err != nil {
+			errmsg, parseErr := parseAsError(msg)
+			if parseErr == nil {
+				return fmt.Errorf("operator %d returned err: %v", i, errmsg)
+			}
+			return err
+		}
+		signedBytes, err := tsp.Message.MarshalSSZ()
+		if err != nil {
+			return err
+		}
+		// Verify that incoming messages have valid DKG ceremony ID
+		if !bytes.Equal(id[:], tsp.Message.Identifier[:]) {
+			return fmt.Errorf("incoming message has wrong ID, aborting... operator %d, msg ID %x", tsp.Signer, tsp.Message.Identifier[:])
+		}
+		// Verification operator signatures
+		if err := c.VerifyFunc(tsp.Signer, signedBytes, tsp.Signature); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Initiator) MakeMultiple(id [24]byte, allmsgs [][]byte) (*wire.MultipleSignedTransports, error) {
+	// We are collecting responses at SendToAll which gives us int(msg)==int(oprators)
+	final := &wire.MultipleSignedTransports{
+		Identifier: id,
+		Messages:   make([]*wire.SignedTransport, len(allmsgs)),
+	}
+	var allMsgsBytes []byte
+	for i := 0; i < len(allmsgs); i++ {
+		msg := allmsgs[i]
+		tsp := &wire.SignedTransport{}
+		if err := tsp.UnmarshalSSZ(msg); err != nil {
+			errmsg, parseErr := parseAsError(msg)
+			if parseErr == nil {
+				return nil, fmt.Errorf("msg %d returned: %v", i, errmsg)
+			}
+			return nil, err
+		}
+		signedBytes, err := tsp.Message.MarshalSSZ()
+		if err != nil {
+			return nil, err
+		}
+		// Verify that incoming messages have valid DKG ceremony ID
+		if !bytes.Equal(id[:], tsp.Message.Identifier[:]) {
+			return nil, fmt.Errorf("incoming message has wrong ID, aborting... operator %d, msg ID %x", tsp.Signer, tsp.Message.Identifier[:])
+		}
+		// Verification operator signatures
+		if err := c.VerifyFunc(tsp.Signer, signedBytes, tsp.Signature); err != nil {
+			return nil, err
+		}
```

Comment:
I moved verification out of this function to VerifyAll, it seems  now we verify twice

Reply:
removed